lib/netext/httpext/request_test.go

```diff
@@ -199,6 +202,20 @@ func TestMakeRequestTimeout(t *testing.T) {
 	require.NoError(t, err)
 	assert.NotNil(t, res)
 	assert.Len(t, samples, 1)
+	sampleCont := <-samples
+	allSamples := sampleCont.GetSamples()
+	require.Len(t, allSamples, 8)
+	expTags := map[string]string{
+		"error":      "context deadline exceeded",
+		"error_code": "1000",
+		"status":     "0",
+		"method":     "GET",
+		"url":        srv.URL,
+		"name":       "",
```

Comment:
Shouldn't this be equal to `url` ?

Reply:
Not according to the `preq.URL.Name != preq.URL.Clean()` check. If I set `Name: srv.URL` on line 196, then the `name` tag is skipped entirely. If I set it to something else, then it does get tagged with the custom value.

Since this is a unit test, I think this is the expected behavior when the `name` tag is not set manually.

Reply:
Yeah ... but it should have `name` ... and it should be equal to the `url`.
Otherwise, it will ... just not have a `name` at all. I would actually expect that in this case `name` isn't set at all

Reply:
Yes, `name` is not set if I set `Name: srv.URL`. Would you prefer that change?

Reply:
I just want every request metric to have `name`, `url` and `method`, just as before ;) 
Currently if the request is with status `0` for some reason and doesn't get to the transport it will not have `name` the same way that this actually fixes this for the other tags (`method` and `url`) it should fix it for `name`

Reply:
> I just want every request metric to have name, url and method, just as before ;)

This PR fixes that. The current behavior of an empty `name` tag (note: not missing, but _empty_) is a different issue which already existed even before #1570. You can confirm this by backporting this test change to e.g. `v0.27.0`.

The issue you fixed in #1634 is separate to this fix, but I think it also makes sense. So let's merge #1634, and then this PR?